romancal/regtest/test_wfi_image_pipeline.py

```diff
@@ -0,0 +1,306 @@
+    diff = compare_asdf(output_filename, truth_filename, **ignore_asdf_paths)
+    assert diff.identical, diff.report()
+
+
+@pytest.mark.soctests
+def test_output_is_image_model(output_model):
+    # DMS280 result is an ImageModel
+    assert isinstance(output_model, rdm.datamodels.ImageModel)
+
+
+@pytest.mark.soctests
+@pytest.mark.parametrize(
+    "step_name",
+    (
+        "assign_wcs",
+        "flat_field",
+        "dark",
+        "dq_init",
+        "jump",
+        "linearity",
+        "ramp_fit",
+        "saturation",
+    ),
+)
+def test_steps_ran(output_model, step_name):
+    # DMS86
+    # also DMS129 for assign_wcs
+    assert getattr(output_model.meta.cal_step, step_name) == "COMPLETE"
+
+
+@pytest.mark.soctests
+def test_jump_in_uneven_ramp(output_model):
+    # DMS361 jump detection detected jumps in uneven ramp
+    uneven = len({len(x) for x in output_model.meta.exposure.read_pattern}) > 1
+    assert uneven & np.any(output_model.dq & pixel.JUMP_DET)
+
+
+@pytest.mark.soctests
+def test_has_a_wcs(output_model):
+    # DMS129  TODO not listed
+    assert output_model.meta.wcs is not None
+
+
+@pytest.mark.soctests
+def test_wcs_has_distortion_information(output_model):
+    # DMS129  TODO not listed
+    assert "v2v3" in output_model.meta.wcs.available_frames
+
+
+@pytest.mark.soctests
+def test_wcs_applies_distortion_correction(output_model):
+    # DMS129  TODO not listed
+    # compare coordinates before and after distortion correction has been applied
+    # 1 - get new image array based on the model
+    x0, y0 = grid_from_bounding_box(output_model.meta.wcs.bounding_box)
+    # 2 - apply the distortion-corrected WCS solution to new image array
+    corrected_coords = output_model.meta.wcs(x0, y0)
+    # 3 - apply the transformation from 'v2v3' to 'world' without distortion correction
+    original_coords = output_model.meta.wcs.get_transform("v2v3", "world")(x0, y0)
+    # compare both results to make sure they don't match
+    # (which means the distortion correction was actually applied to the model)
+    assert (corrected_coords[0] != original_coords[0]).all()
+    assert (corrected_coords[1] != original_coords[1]).all()
+
+
+@pytest.mark.soctests
+@pytest.mark.parametrize(
+    "arr_name", ("dq", "err", "var_poisson", "var_rnoise", "var_flat")
+)
+def test_array_exists(output_model, arr_name):
+    # DMS87
+    assert hasattr(output_model, arr_name)
+
+
+@pytest.mark.soctests
+def test_has_exposure_time(output_model):
+    # DMS88 total exposure time exists
+    assert "exposure_time" in output_model.meta.exposure
+
+
+@pytest.mark.soctests
+@pytest.mark.parametrize("meta_attribute", ("detector", "optical_element"))
+def test_instrument_meta(output_model, meta_attribute):
+    # DMS-136 PSF tests  TODO not listed
+    assert meta_attribute in output_model.meta.instrument
+
+
+@pytest.mark.soctests
+def test_wcs_has_bounding_box(output_model):
+    # DMS89 WCS tests  TODO not listed
```

Comment:
DMS89 is not listed in the requirements. Is it satisfied?

Reply:
This is likewise a requirement that should be added to the requirements.json list; we've delivered this requirement and this was the test that did that.

Reply:
Added DMS89 in 03ce0ad